campbot/checkers.py

```diff
@@ -475,3 +477,62 @@ def __call__(self, contrib, old_version, new_version):
             return True, True
 
         return True, old_doc.quality == new_doc.quality
+    
+class MissingAccess():
+    """
+    Route with no access
+    """
+    def __init__(self):
+        self.name = "Accès Manquant"
+        self.fail_marker = emoji("/images/emoji/apple/red_circle.png?v=3", self.name)
+        self.success_marker = ""
+
+    def __call__(self, contrib, old_version, new_version):
+        def test(version):
+            if not version:
+                return True
+
+            if "redirects_to" in version.document:
+                return True
+
+            if version.document.type != "r":
+                return True
+
+            waypoints = version.document.associations.waypoints
+            bool_value = False
+            for waypoint in waypoints:
+                if waypoints.waypoint_type = "access":
+                    bool_value=True
+                    
+            return bool_value
+
+        return True, test(new_version)
```

Comment:
Ok, quite an obscur point on those tests. The return value of this function is two booleans:

1. the old version was good ?
2. the new version was good ?

BUT test can be 1 of these 2 categories : 

1. Test a given version (for instance, route must contains access), and report the contribution if the old version was ok AND the new one is not ok.
2. Test the transition, for instance, do the contributor move the geoloc more than N meters.

When you are in the second use case, the first boolean is not used, and you must set it to `true`. But, if your are in the first use case, you must test the old version, otherwise, the contribution will be reported, even if the contributor did a modification without any link to the tested point.

```suggestion
        return test(old_version), test(new_version)
```